### PR TITLE
MULE-10435: Add a way to shared libs on the isolated tests

### PR DESCRIPTION
**Pull request by gsfernandes**

### CODE REVIEW

File: tests/runner/pom.xml [hunk 1]
```diff
@@ -47,43 +47,60 @@
...
         </dependency>
-
         <dependency>
             <groupId>org.mule.tests</groupId>
             <artifactId>mule-tests-unit</artifactId>
             <version>${project.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
```
Comment (pablokraan) [inline tests/runner/pom.xml]: Email extension is also using this, it should be defined in mule as managed dependency

File: tests/runner/src/test/java/org/mule/test/runner/api/AetherClassPathClassifierTestCase.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  private Dependency muleArtifactDep;
+  private Dependency muleTestFunctionalDep;
+  private Dependency guavaDep;
+  private Dependency derbyDriverDep;
+
+  private List<Dependency> directDependencies;
+
+  @Before
+  public void before() throws Exception {
+    this.rootArtifact = new DefaultArtifact("org.mule.tests:mule-tests-runner:4.0-SNAPSHOT");
```
Comment (pablokraan) [inline tests/runner/src/test/java/org/mule/test/runner/api/AetherClassPathClassifierTestCase.java]: Is the 4.0-SNAPSHOT related to the Mule project version?

File: tests/runner/src/test/java/org/mule/test/runner/api/AetherClassPathClassifierTestCase.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+  private ArtifactDescriptorResult noManagedDependencies() throws ArtifactDescriptorException {
+    ArtifactDescriptorResult artifactDescriptorResult = mock(ArtifactDescriptorResult.class);
+    List<Dependency> managedDependencies = Collections.<Dependency>emptyList();
+    when(artifactDescriptorResult.getManagedDependencies()).thenReturn(managedDependencies);
+    when(dependencyResolver.readArtifactDescriptor(any(Artifact.class))).thenReturn(artifactDescriptorResult);
+    return artifactDescriptorResult;
+  }
+
+  class ListOfElements extends ArgumentMatcher<List> {
```
Comment (pablokraan) [inline tests/runner/src/test/java/org/mule/test/runner/api/AetherClassPathClassifierTestCase.java]: cannot use IsCollectionWithSize matcher?

File: tests/runner/src/test/java/org/mule/test/runner/api/AetherClassPathClassifierTestCase.java (hunk 2 above)
Comment (gsfernandes) [inline tests/runner/src/test/java/org/mule/test/runner/api/AetherClassPathClassifierTestCase.java]: No, it is just mock data.

Comment (pablokraan) [inline tests/runner/src/test/java/org/mule/test/runner/api/AetherClassPathClassifierTestCase.java]: the I would suggest you to use fantasy artifact IDs, like org.foo:foo:1.0

**Pull request merged**